src/apps/HeyGPT/HeyGPT/ViewModels/LetsChatViewModel.cs

```diff
@@ -195,5 +280,133 @@ private void AddErrorMessageToHistory(string errorMessage)
         });
     }
 
+    private async Task SendMessageInternal(string userMessage, ChatCharacterDetails? desiredCharacterDetails = null)
```

Comment:
This code was in the command handler for the send message... Now we have a private method used in three places